moved/src/types/state.rs

```diff
@@ -378,6 +380,76 @@ fn compute_from(tx: &OpTxEnvelope) -> Result<Address, alloy::primitives::Signatu
 #[derive(Debug)]
 pub struct TransactionResponse(pub op_alloy::rpc_types::Transaction);
 
+impl TransactionResponse {
+    pub fn canonical(
+        inner: OpTxEnvelope,
+        block_hash: B256,
+        block_number: u64,
+        transaction_index: u64,
+        effective_gas_price: u128,
+        from: Address,
+    ) -> Self {
+        Self(op_alloy::rpc_types::Transaction {
+            inner: alloy::rpc::types::eth::Transaction {
+                inner,
+                block_hash: Some(block_hash),
+                block_number: Some(block_number),
+                transaction_index: Some(transaction_index),
+                effective_gas_price: Some(effective_gas_price),
+                from,
+            },
+            deposit_nonce: None,
+            deposit_receipt_version: None,
+        })
+    }
+
+    pub fn deposit(
+        inner: OpTxEnvelope,
+        block_hash: B256,
+        block_number: u64,
+        transaction_index: u64,
+        effective_gas_price: u128,
+        from: Address,
+        deposit_nonce: u64,
+        deposit_receipt_version: u64,
+    ) -> Self {
+        Self(op_alloy::rpc_types::Transaction {
+            inner: alloy::rpc::types::eth::Transaction {
+                inner,
+                block_hash: Some(block_hash),
+                block_number: Some(block_number),
+                transaction_index: Some(transaction_index),
+                effective_gas_price: Some(effective_gas_price),
+                from,
+            },
+            deposit_nonce: Some(deposit_nonce),
+            deposit_receipt_version: Some(deposit_receipt_version),
+        })
+    }
+}
+
+impl From<Transaction> for TransactionResponse {
+    fn from(value: Transaction) -> Self {
+        let (deposit_nonce, deposit_receipt_version) = get_deposit_nonce(&value.0)
+            .map(|nonce| (Some(nonce.nonce), Some(nonce.version)))
+            .unwrap_or((None, None));
+
+        Self(op_alloy::rpc_types::Transaction {
+            inner: alloy::rpc::types::eth::Transaction {
+                inner: value.0,
+                // TODO: Solve missing fields
+                block_hash: None,
```

Comment:
Are these really not available? We should really figure these out, cause when we were debugging the deposit issue the `None` values caused so much headache.